docs/en/operations/analyzer.md

```diff
@@ -0,0 +1,104 @@
+---
+slug: /en/operations/analyzer
+sidebar_label: Analyzer
+---
+
+# Analyzer
+
+<BetaBadge />
+
+## Known incompatibilities
+
+In ClickHouse version `24.3`, new query analysis was enabled by default.
+Despite fixing a large number of bugs and introducing new optimizations, it also introduces some breaking changes in ClickHouse behaviour.
+
+### Invalid before optimization queries
+
+The previous query planning infrastructure applied AST-level optimizations before the query validation step.
+Optimizations could rewrite the initial query so it becomes valid and can be executed.
+
+In the new infrastructure, query validation takes place before the optimization step.
+This means that invalid queries that were possible to execute before are now unsupported.
+
+**Example 1:**
+
+```sql
+SELECT number
+FROM numbers(1)
+GROUP BY toString(number)
+```
+
+The following query uses column `number` in the projection list when only `toString(number)` is available after the aggregation.
+In the old infrastructure, `GROUP BY toString(number)` was optimized into `GROUP BY number,` making the query valid.
+
+**Example 2:**
+
+```sql
+SELECT
+    number % 2 AS n,
+    sum(number)
+FROM numbers(10)
+GROUP BY n
+HAVING number > 5
+```
+
+The same problem occurs in this query: column `number` is used after aggregation with another key.
+The previous query analyzer fixed this query by moving the `number > 5` filter from the `HAVING` clause to the `WHERE` clause.
+
+### Known incompatibilities of JOIN clause
+
+* Using expression from `SELECT` list in `JOIN` key as an expression from LEFT table. Example.  Fix (best effort, should be under compatibility flag).
+* Similar issue ^. Alias for column (in select list) now applied to JOIN result (and not to left table). Example from Denny Crane. New behavior is the correct one. Will try to add best-effort compatibility setting.
+* Columns names are changed for some queries. This might breaks some scripts.  Example.
```

Comment:
Looks like example from `Columns names are changed for some queries` is fixed, at least mentioned example https://fiddle.clickhouse.com/b4b4a7b1-67e6-47f7-a3a8-a2ef3095679a in `head` returns the same result.

Reply:
For first two point I added info to this branch, because it's impossible to create suggestions in PR with code inside